gpiozero/pins/native.py

```diff
@@ -324,7 +425,10 @@ def __init__(self):
         self.mem = GPIOMemory(self.pi_info.soc)
         self.fs = GPIOFS(self, queue)
         self.dispatch = NativeDispatchThread(self, queue)
-        self.pin_class = NativePin
+        if self.pi_info.soc == 'BCM2711':
+            self.pin_class = Native2711Pin
+        else:
+            self.pin_class = Native2835Pin
```

Comment:
So I guess this means that `NativePin` itself will never be directly used? Would it be worth putting in an exception or something to enforce that?

Reply:
Yes, NativePin is effectively an abstract class at this point. I've never been hugely fussed about banning the instantiation of abstract classes; it's one of those issues that's come up sufficiently rarely and (provided abstract methods are marked with some appropriate error like NotImplementedError) is pretty obvious if/when it does that it's not a headache to track down. In this case I think I'll leave it - I don't particularly want to complicate the hierarchy at the pin level with Python's abstractclass machinery.

Reply:
Should `_get_pull` and `_set_pull` "base methods" in `NativePin` raise `NotImplementedError` ?

Reply:
`_set_pull` effectively does by inheritance (admittedly not NotImplementedError, I think it's PinFixedPull, but still that'd be a pretty big hint to anyone running into it that an override for `_set_pull` was missing. But yes, `_get_pull` is missing one (it'll just return 'floating' by default). Yup, you've convinced me: I should chuck a `NotImplementedError` into those (won't cost anything at runtime as the super-class should never get called for those). Will do that in a mo.